dattri/algorithm/influence_function.py

```diff
@@ -433,30 +433,200 @@ def transformation_on_query(
             torch.Tensor: The transformation on the query. Normally it is a 2-d
                 dimensional tensor with the shape of (batchsize, transformed_dimension).
         """
-        from dattri.func.fisher import ifvp_datainf
 
-        model_params, param_layer_map = self.task.get_param(index, layer_split=True)
+        def _single_datainf(
+            v: torch.Tensor,
+            grad: torch.Tensor,
+            regularization: float,
+        ) -> torch.Tensor:
+            """Intermediate DataInf value calculation of a single data point.
```

Comment:
More detailed description. Better include the markdown format of the formula of this function